ui/src/app/pages/analysis/runtime-panel.tsx

```diff
@@ -323,6 +458,25 @@ export const RuntimePanel = fp.flow(withCurrentWorkspace(), withCdrVersions())((
         }
       </FlexColumn>
     </div>
+    {runtimeNeedsRestart && <FlexRow
+        style={{
+          alignItems: 'center',
+          backgroundColor: colorWithWhiteness(colors.warning, .9),
+          border: `1px solid ${colors.warning}`,
+          borderRadius: '5px',
+          color: colors.dark,
+          marginTop: '.5rem',
+          padding: '.5rem 0px'
+        }}
+    >
+      <ClrIcon
+          style={{color: colors.warning, marginLeft: '.5rem'}}
+          shape={'warning-standard'}
+          size={16}
+          class={'is-solid'}
+      />
+      <div style={{marginLeft: '.5rem'}}>You've made changes that require recreating your environment to take effect.</div>
```

Comment:
Thanks for adding this. Technically this should probably be in @ericsong 's UI PR, since currently all changes induce restarts until we have Update integration - but I'm fine adding it here since you already did the styling.

The logic for `runtimeNeedsRestart` is also not quite correct, e.g. shrinking the disk size is illegal via Update, and will require a restart. I don't know if that's the only issue, part of the update ticket was to identify exactly which config changes are supported.

Alternatively, just drop `runtimeNeedsRestart` and only show this dialog with `runtimeChanged` - which would currently be accurate.

Reply:
will do the latter